lib/web/app/fragment.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/httplib"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/utils"
+	"github.com/gravitational/trace"
+
+	"github.com/julienschmidt/httprouter"
+)
+
+type fragmentRequest struct {
+	CookieValue string `json:"cookie_value"`
+}
+
+func (h *Handler) handleFragment(w http.ResponseWriter, r *http.Request, p httprouter.Params) error {
+	switch r.Method {
+	case http.MethodGet:
+		nonce, err := utils.CryptoRandomHex(auth.TokenLenBytes)
+		if err != nil {
+			h.log.Debugf("Failed to generate and encode random numbers: %v.", err)
```

Comment:
Shouldn't this be logged at error level?

Reply:
@russjones `h.log.WithError(err).Debugf("...")`